src/detect-template-buffer.c

```diff
@@ -48,6 +48,9 @@ static int DetectTemplateBufferSetup(DetectEngineCtx *de_ctx, Signature *s,
 {
     s->list = DETECT_SM_LIST_TEMPLATE_BUFFER_MATCH;
     s->alproto = ALPROTO_TEMPLATE;
+    s->flags |= SIG_FLAG_STATE_MATCH;
+    s->mask |= SIG_MASK_REQUIRE_TEMPLATE_STATE;
```

Comment:
Just keep in mind that to make a detect module dynamically loadable, its setup code will need to consolidated. Not to mention, its easier to document how to make a new detect module.

Yes, it would be best to do for all at once, this was more to show the idea.